modules/apps/marketplace/marketplace-js-components-web/test/js/components/Carousel.spec.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import '@testing-library/jest-dom/extend-expect';
+import {fireEvent, render} from '@testing-library/react';
+import React from 'react';
+
+import Carousel from '../../../src/main/resources/META-INF/resources/js/components/Carousel';
+
+const mockPrev = jest.fn();
+
+const mockNext = jest.fn();
+
+const mockSelect = jest.fn();
+
+const imagesURL = [
+	'https://liferay.com/liferay-logo.png',
+	'https://liferay.com/liferay-ray.png',
+	'https://liferay.com/liferay-waffle.png',
+];
+
+describe('Carousel', () => {
+	it('rendering components withou clicking button', () => {
+		const {container} = render(<Carousel images={imagesURL} />);
+
+		const images = container.querySelectorAll('img');
+
+		const previousButton = container.querySelector(
+			'.left'
+		) as HTMLButtonElement;
+		previousButton?.addEventListener('click', mockPrev);
+
+		const nextButton = container.querySelector(
+			'.right'
+		) as HTMLButtonElement;
+		nextButton?.addEventListener('click', mockNext);
+
+		const selectButton = container.querySelector(
+			'.gallery-image'
+		) as HTMLButtonElement;
+		selectButton?.addEventListener('click', mockSelect);
+
+		const slide = container.querySelector('.carousel-image');
+
+		expect(slide).toHaveAttribute('alt', `Slide 0`);
+		expect(images[0]).toHaveAttribute('src', `${imagesURL[0]}`);
+		expect(images[1]).toHaveAttribute('src', `${imagesURL[0]}`);
+		expect(images[2]).toHaveAttribute('src', `${imagesURL[1]}`);
+		expect(images[3]).toHaveAttribute('src', `${imagesURL[2]}`);
```

Comment:
is it possible to move them closer to images declaration or move images declaration closer to them?

something like:

const images = container.querySelectorAll('img');

expect(images[0]).toHaveAttribute('src', `${imagesURL[0]}`);
expect(images[1]).toHaveAttribute('src', `${imagesURL[0]}`);
expect(images[2]).toHaveAttribute('src', `${imagesURL[1]}`);
expect(images[3]).toHaveAttribute('src', `${imagesURL[2]}`);